### PR TITLE
Add Course::LessonPlanItem model

### PR DESCRIPTION
**Pull request by kxmbrian**
This PR is for the lesson plan item model as per https://github.com/Coursemology/coursemology2/wiki/EXP-Points-and-Levels.

### CODE REVIEW

File: spec/models/course/lesson_plan_item_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+RSpec.describe Course::LessonPlanItem, type: :model do
+  it { is_expected.to validate_numericality_of(:base_exp).only_integer }
+  it { is_expected.to validate_numericality_of(:time_bonus_exp).only_integer }
+  it { is_expected.to validate_numericality_of(:extra_bonus_exp).only_integer }
+
+  let!(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    let(:lesson_plan_item) { FactoryGirl.create :course_lesson_plan_item }
+    context '.total_exp' do
```
Comment (allenwq) [inline spec/models/course/lesson_plan_item_spec.rb]: should be <code>describe '#total_exp' </code>

And can you also add a test case for <code>acts_as_lesson_plan_item</code> ?

Comment (kxmbrian) [inline spec/models/course/lesson_plan_item_spec.rb]: `acts_as_lesson_plan_item` will come soon in another PR!

File: app/models/course/lesson_plan_item.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  actable
+  stampable
+  validates :base_exp, numericality: { only_integer: true }
+  validates :time_bonus_exp, numericality: { only_integer: true }
+  validates :extra_bonus_exp, numericality: { only_integer: true }
+
+  after_initialize :set_default_values
+
+  def total_exp
+    base_exp + time_bonus_exp + extra_bonus_exp
```
Comment (allenwq) [inline app/models/course/lesson_plan_item.rb]: sorry, I think what I've commented in #177 should be applied here.

**Comment by kxmbrian** [comment]
Replaced by #184. Changes:
- Pulled from a namespaced branch
- PR is now dependent on #182

**Pull request closed without merging**